app/helpers/layout_helper.rb

```diff
@@ -9,4 +9,15 @@ def body_class
     user_signed_in? ? classes << "user-signed_in" : classes << "user-signed_out"
     classes.join(" ")
   end
+
+  def js_packs
+    packs = ['application']
+    packs << 'internal' if user_signed_in?
+    if (namespace_name == "tracks" && controller_name == "exercises" && action_name == "edit") ||
+       (namespace_name == "test" && controller_name == "editor" && action_name == "show")
+      packs << 'editor'
+    end
```

Comment:
I'd be tempted to move the condition in a separate method, something like:

```suggestion
    packs << 'editor' if editor_displayed?
```